CorsixTH/Lua/dialogs/bottom_panel.lua

```diff
@@ -206,29 +206,59 @@ end
 function UIBottomPanel:drawDynamicInfo(canvas, x, y)
   if self.world:isCurrentSpeed("Pause") and not self.world.user_actions_allowed then
     self.pause_font:drawWrapped(canvas, _S.misc.pause, x + 10, y + 14, 255, "center")
-  elseif self.dynamic_info then
-    local info = self.dynamic_info
-    local font = self.white_font
-    for i, text in ipairs(info["text"]) do
-      font:drawWrapped(canvas, text, x + 20, y + 10 * i, 240)
-      if i == #info["text"] and info["progress"] then
-        local white = canvas:mapRGB(255, 255, 255)
-        local black = canvas:mapRGB(0, 0, 0)
-        local orange = canvas:mapRGB(221, 83, 0)
-        canvas:drawRect(white, x + 165, y + 10 * i, 100, 10)
-        canvas:drawRect(black, x + 166, y + 1 + 10 * i, 98, 8)
-        canvas:drawRect(orange, x + 166, y + 1 + 10 * i, math.floor(98 * info["progress"]), 8)
-        if info["dividers"] then
-          for _, value in ipairs(info["dividers"]) do
-            canvas:drawRect(white, x + 165 + math.floor(value * 100), y + 10 * i, 1, 10)
-          end
+    return
+  end
+
+  if not (self.dynamic_info and self.dynamic_info["text"]) then
+    return
+  end
+
+  local info = self.dynamic_info
+  local font = self.white_font
+  for i, text in ipairs(info["text"]) do
+    font:drawWrapped(canvas, text, x + 20, y + 10 * i, 240)
+    if i == #info["text"] and info["progress"] then
```

Comment:
Seems a bit weird to special-check the last entry, and not even use either `i` or `text`, unless I am missing something.

Reply:
I didn't refactor that but it could certainly be pulled out of the loop. They are just drawing any progress bar after all text.